src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -73,36 +73,40 @@ public class IndexStore : IDisposable
                                                  IndexCompiler.IndexExtension.Length -
                                                  4; // ".dll"
 
-        public readonly DatabaseIndexLockModeController LockMode;
+        private readonly DocumentDatabase _documentDatabase;
 
-        public readonly DatabaseIndexPriorityController Priority;
+        private readonly CollectionOfIndexes _indexes = new CollectionOfIndexes();
+        private readonly ConcurrentDictionary<string, SemaphoreSlim> _indexLocks = new ConcurrentDictionary<string, SemaphoreSlim>(StringComparer.OrdinalIgnoreCase);
 
-        public readonly DatabaseIndexStateController State;
+        private bool _initialized;
 
-        public readonly DatabaseIndexDeleteController Delete;
+        private bool _run = true;
 
-        public readonly DatabaseIndexCreateController Create;
+        private long _lastSurpassedAutoIndexesDatabaseRecordEtag;
 
-        public readonly DatabaseIndexHasChangedController HasChanged;
+        private const int PathLengthLimit = 259; // Roslyn's MetadataWriter.PathLengthLimit = 259
 
-        public IndexStore(DocumentDatabase documentDatabase, ServerStore serverStore)
+        protected IndexStore(DocumentDatabase documentDatabase)
         {
             _documentDatabase = documentDatabase;
-            _serverStore = serverStore;
 
             LockMode = new DatabaseIndexLockModeController(documentDatabase);
             Priority = new DatabaseIndexPriorityController(documentDatabase);
             State = new DatabaseIndexStateController(documentDatabase);
             Delete = new DatabaseIndexDeleteController(documentDatabase);
-            Create = new DatabaseIndexCreateController(documentDatabase);
             HasChanged = new DatabaseIndexHasChangedController(documentDatabase);
-
             Logger = LoggingSource.Instance.GetLogger<IndexStore>(_documentDatabase.Name);
 
             var stoppedConcurrentIndexBatches = _documentDatabase.Configuration.Indexing.NumberOfConcurrentStoppedBatchesIfRunningLowOnMemory;
             StoppedConcurrentIndexBatches = new SemaphoreSlim(stoppedConcurrentIndexBatches);
         }
 
+        public IndexStore(DocumentDatabase documentDatabase, ServerStore serverStore) : this(documentDatabase)
+        {
+            ServerStore = serverStore;
+            Create = new DatabaseIndexCreateController(documentDatabase);
+        }
```

Comment:
I think we should have here a single `ctor`

Reply:
I think the 2 ctors is the nicest way to handle `DatabaseIndexCreateController` instantiation with the appropriate type 
what would you prefer? to have a virtual call inside ctor `CreateDatabaseIndexCreateController` or something like that ?